### DISCUSSION TITLE
Workflow not triggering (nor asking for permission) on external Pull Request

### DISCUSSION THREAD

**Discussion by David-Lor** [community/community#36561] [Actions] [answered]
Somebody [opened a Pull Request](https://github.com/David-Lor/pnytter/pull/6) in one of my public repositories, where I have a [Github Workflow for running tests](https://github.com/David-Lor/pnytter/blob/develop/.github/workflows/test.yaml). However, the opened pull request doesn't show this workflow on the Checks list, so I cannot give it permissions to run on this PR.

These are the Actions settings within this repository:
![imagen](https://user-images.githubusercontent.com/17401854/196281746-9c7c6bbb-cf7b-4f88-8771-da68bb000f45.png)

According to the [docs](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) regarding workflows execution on externally-opened pull requests, I should see the workflow as a check with the possibility of allowing its execution, but no checks are visible on the mentioned pull request.

The [workflow](https://github.com/David-Lor/pnytter/blob/develop/.github/workflows/test.yaml) has a `push` filter by paths, but the mentioned PR has changes within the defined paths, so it should run. Changes made by myself on [other pull request](https://github.com/David-Lor/pnytter/pull/9) trigger the workflow. Is there something I'm missing here?

**Comment by airtower-luna** [comment, marked as answer]
The problem is that the workflow is using a `push` trigger, not `pull_request`. The `push` event triggers a workflow *in the repository where the branch is*, if actions are enabled there (by default they aren't in forks). That's why it works with your own PR, which is in the same repository.

The best solution is to add a `pull_request` trigger to the workflow, and ask the contributor to rebase or merge. Unfortunately having both `pull_request` and `push` will cause duplicate runs for your own PRs (first for the push, then on creating/updating the PR), consider if you want to accept that or limit the `push` trigger to specific branches.

**Reply by David-Lor** [reply]
Alright! I didn't know about that. Maybe with some `if` magic I can avoid running the workflow twice. Thanks for the hint!

**Comment by ossanna16** [comment]
Hey there @David-Lor and thank you for asking a good question! I'm glad you found our community 🙂
